src/main/java/stirling/software/SPDF/config/RuntimePathConfig.java

```diff
@@ -0,0 +1,76 @@
+package stirling.software.SPDF.config;
+
+import java.io.File;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.context.annotation.Configuration;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import stirling.software.SPDF.model.ApplicationProperties;
+import stirling.software.SPDF.model.ApplicationProperties.CustomPaths.Operations;
+import stirling.software.SPDF.model.ApplicationProperties.CustomPaths.Pipeline;
+
+@Slf4j
+@Configuration
+@Getter
+public class RuntimePathConfig {
+    private final ApplicationProperties properties;
+    private final String basePath;
+    private final String weasyPrintPath;
+    private final String unoConvertPath;
+
+    // Pipeline paths
+    private final String pipelineWatchedFoldersPath;
+    private final String pipelineFinishedFoldersPath;
+    private final String pipelineDefaultWebUiConfigs;
+    private final String pipelinePath;
+
+    public RuntimePathConfig(ApplicationProperties properties) {
+        this.properties = properties;
+        this.basePath = InstallationPathConfig.getPath();
+
+        String pipelinePath = basePath + "pipeline" + File.separator;
+        String watchedFoldersPath = pipelinePath + "watchedFolders" + File.separator;
+        String finishedFoldersPath = pipelinePath + "finishedFolders" + File.separator;
+        String webUiConfigsPath = pipelinePath + "defaultWebUIConfigs" + File.separator;
+
+        Pipeline pipeline = properties.getSystem().getCustomPaths().getPipeline();
+        if (pipeline != null) {
+            if (!StringUtils.isEmpty(pipeline.getWatchedFoldersDir())) {
+                watchedFoldersPath = pipeline.getWatchedFoldersDir();
+            }
+            if (!StringUtils.isEmpty(pipeline.getFinishedFoldersDir())) {
+                finishedFoldersPath = pipeline.getFinishedFoldersDir();
+            }
+            if (!StringUtils.isEmpty(pipeline.getWebUIConfigsDir())) {
+                webUiConfigsPath = pipeline.getWebUIConfigsDir();
+            }
+        }
+
+        this.pipelinePath = pipelinePath;
+        this.pipelineWatchedFoldersPath = watchedFoldersPath;
+        this.pipelineFinishedFoldersPath = finishedFoldersPath;
+        this.pipelineDefaultWebUiConfigs = webUiConfigsPath;
+
+        // Initialize Operation paths
+        String weasyPrintPath = "/opt/venv/bin/weasyprint";
```

Comment:
It is therefore not usable for Windows users

`weasyprint` is available for Windows

Reply:
thats why i have added this feature, do allow users to define their own custom paths,
I guess I can change defaults though for different OS's.. let me add

Reply:
Updated to use PATH as default for all systems non docker